pkg/server/commands/reverseexpand/reverse_expand.go

```diff
@@ -173,36 +183,38 @@ func NewResolutionMetadata() *ResolutionMetadata {
 }
 
 // Execute yields all the objects of the provided objectType that the given user has, possibly, a specific relation with
-// and sends those objects to resultChan. It MUST guarantee no duplicate objects sent.
+// and sends those objects to the input channel. It MUST guarantee no duplicate objects sent.
+// The input channel can be buffered or not. If it's not buffered, this function will block when it finds either a candidate object or an error.
 //
-// If an error is encountered before resolving all objects: the provided channel will NOT be closed and
+// If an error is encountered before resolving all objects:
 // - if the error is context cancellation or deadline: Execute may send the error through the channel
 // - otherwise: Execute will send the error through the channel
-// If no errors, Execute will yield all of the objects on the provided channel and then close the channel
-// to signal that it is done.
+// If no errors, Execute will yield all of the objects on the provided channel.
+// In all cases, at the end, the input channel is closed.
 func (c *ReverseExpandQuery) Execute(
 	ctx context.Context,
 	req *ReverseExpandRequest,
-	resultChan chan<- *ReverseExpandResult,
+	reverseExpandResultsChan chan<- *ReverseExpandResult,
 	resolutionMetadata *ResolutionMetadata,
 ) {
-	err := c.execute(ctx, req, resultChan, false, resolutionMetadata)
+	err := c.execute(ctx, req, reverseExpandResultsChan, false, resolutionMetadata)
 	if err != nil {
 		select {
 		case <-ctx.Done():
```

Comment:
Why not handle this case the same as the one below? It's the callers choice how to handle cancellations of the underlying ReverseExpand call.

e.g.

```go
if err != nil {
    select {
        case err := <-ctx.Done()
            reverseExpandResultsChan <- &ReverseExpandResult{Err: err}
        case reverseExpandResultsChan <- &ReverseExpandResult{Err: err}
    }
}
```

Usually `ctx.Err()` isn't swallowed or treated differently than `err != nil` cases, so it's odd we're swallowing that specifically.

Reply:
We need to consider the scenario that the channel might be full already, so trying to send anything to it will block indefinitely

Reply:
Update: before this PR we had

```go
if err != nil {
		select {
		case <-ctx.Done():
			return
		case resultChan <- &ReverseExpandResult{Err: err}:
			return
		}
	}
```

I noticed while writing a new unit test that, unless you call `cancel()` on the context, this will block forever too, which means the test never ends.

So I changed it to

```go
if err != nil {
		select {
		case resultChan <- &ReverseExpandResult{Err: err}:
		default:
		}
	}
```

so that we don't need context cancellation for ReverseExpand to finish.

Reply:
Updated the PR description with more details